File: Tests/Tests/STPEphemeralKeyTest.m
//
//  STPEphemeralKeyTest.m
//  Stripe
//
//  Created by Ben Guo on 5/17/17.
//  Copyright © 2017 Stripe, Inc. All rights reserved.
//

#import <XCTest/XCTest.h>
#import "STPEphemeralKey.h"
#import "STPTestUtils.h"

@interface STPEphemeralKeyTest : XCTestCase

@end

@implementation STPEphemeralKeyTest

- (void)testDecoding {
    NSDictionary *json = [STPTestUtils jsonNamed:@"EphemeralKey"];
    STPEphemeralKey *key = [STPEphemeralKey decodedObjectFromAPIResponse:json];
    XCTAssertNotNil(key);
    XCTAssertEqualObjects(key.stripeID, json[@"id"]);
    XCTAssertEqualObjects(key.secret, json[@"secret"]);
    XCTAssertEqualObjects(key.created, [NSDate dateWithTimeIntervalSince1970:[json[@"created"] doubleValue]]);
    XCTAssertEqualObjects(key.expires, [NSDate dateWithTimeIntervalSince1970:[json[@"expires"] doubleValue]]);
    XCTAssertEqual(key.livemode, [json[@"livemode"] boolValue]);
    XCTAssertEqualObjects(key.customerID, @"cus_123");
}

@end

File: Tests/Tests/STPEphemeralKeyTest.swift
//
//  STPEphemeralKeyTest.swift
//  Stripe
//
//  Created by Ben Guo on 5/17/17.
//  Copyright © 2017 Stripe, Inc. All rights reserved.
//

@testable import Stripe

class STPEphemeralKeyTest: XCTestCase {
  func testDecoding() {
    let json = STPTestUtils.jsonNamed("EphemeralKey")!
    let key = STPEphemeralKey.decodedObject(fromAPIResponse: json)!
    XCTAssertEqual(key.stripeID, json["id"] as! String)
    XCTAssertEqual(key.secret, json["secret"] as! String)
    XCTAssertEqual(
      key.created,
      Date(timeIntervalSince1970: TimeInterval((json["created"] as! NSNumber).doubleValue)))
    XCTAssertEqual(
      key.expires,
      Date(timeIntervalSince1970: TimeInterval((json["expires"] as! NSNumber).doubleValue)))
    XCTAssertEqual(key.livemode, (json["livemode"] as! NSNumber).boolValue)
    XCTAssertEqual(key.customerID, "cus_123")
  }
}
